### DISCUSSION TITLE
Y axis tick interval

### DISCUSSION THREAD

**Discussion by RahmanMonjur** [TanStack/react-charts#189] [General] [answered]
Hi experts et @tannerlinsley ,
Is there any way to set intervals for Y-axis ticks? For example, I want to set 0,20,40,60,80, and 100 on the Y-axis.
Thanks in advance.
Monjur

**Comment by tannerlinsley** [comment, marked as answer]
Due to a massive overhaul and rewrite of React Charts to TypeScript and a massively upgraded public API, this discussion is being marked as answered for triage. The latest version can be tested via the `react-charts@beta` tag. Documentation is in the process of being upgraded, but thankfully, everything is fully typed now, so you can explore the API that way for now.

If, *after testing or playing with the v3 beta*, you believe this discussion is still relevant or unanswered, feel free to mark it as such and continue the discussion.

**Reply by GreenAsJade** [reply]
I can't see, in the API or Axis type definitions, any way to set the tick intervals.

I can see a way to set the tickCount, but this isn't the same, because (for the Y axis in particular) you'd need to know the min and max value of your data to use tick count to get tick interval

**Reply by SyedFaranMustafa** [reply]
```
export declare type AxisOptionsBase = {
    primaryAxisId?: string;
    elementType?: 'line' | 'area' | 'bar' | 'bubble';
    showDatumElements?: boolean | 'onFocus';
    curve?: CurveFactory;
    invert?: boolean;
    position?: Position;
    minTickPaddingForRotation?: number;
    tickLabelRotationDeg?: number;
    tickCount?: number;
    shouldNice?: boolean;
    innerBandPadding?: number;
    outerBandPadding?: number;
    innerSeriesBandPadding?: number;
    outerSeriesBandPadding?: number;
    minBandSize?: number;
    maxBandSize?: number;
    minDomainLength?: number;
    showGrid?: boolean;
    show?: boolean;
    stacked?: boolean;
    stackOffset?: typeof stackOffsetNone;
    id?: string | number;
    styles?: CSSProperties & {
        line?: CSSProperties;
        tick?: CSSProperties;
    };

```

These parameters for the option I don't think there is tick interval in it
